### PR TITLE
Verify location and save in submission object

### PR DESCRIPTION
**Pull request by kmcnellis**
Also update the map to draw polygons

### CODE REVIEW

File: client/components/eventform/eventform.controller.js [hunk 1]
```diff
@@ -168,32 +191,81 @@ angular.module('venueApp')
       $scope.selectingEvent = false;
       $scope.eventInfo = event;
     };
+          // var bounds = map.getBounds();
+          // var ne = bounds.getNorthEast();
+          // var sw = bounds.getSouthWest();
+
```
Comment (seveibar) [inline client/components/eventform/eventform.controller.js]: scary ghost code

File: client/components/eventform/eventform.controller.js [hunk 2]
```diff
@@ -168,32 +191,81 @@ angular.module('venueApp')
...
+            description: $scope.event.location.description,
             geo: {
               type: 'Point',
               coordinates: [
                 $scope.event.map.center.longitude,
                 $scope.event.map.center.latitude
               ]
             },
-            radius: Math.abs($scope.event.map.bounds.northeast.longitude-$scope.event.map.bounds.southwest.longitude)
+            radius: Math.abs($scope.event.map.bounds.northeast.longitude-$scope.event.map.bounds.southwest.longitude),
```
Comment (seveibar) [inline client/components/eventform/eventform.controller.js:294]: hmm, wouldn't

`Math.max(Math.abs($scope.event.map.bounds.northeast.longitude-$scope.event.map.bounds.southwest.longitude),Math.abs($scope.event.map.bounds.northeast.latitude-$scope.event.map.bounds.southwest.latitude))` be a bit better?

File: server/api/submission/submission.controller.js [hunk 3]
```diff
@@ -226,26 +227,52 @@ exports.show = function(req, res) {
...
-    var submit = {
-      images : imagePaths,
-      submitter : req.body.userId,
-      authors : req.body.authors,
-      sectionEvent : req.body.eventId,
-      location : {
-        geo: {
+    req.body.coordinates[0] = Number(req.body.coordinates[0]);
+    req.body.coordinates[1] = Number(req.body.coordinates[1]);
+    if (!req.body.authors){
```
Comment (seveibar) [inline server/api/submission/submission.controller.js:237]: Coordinates are not mandatory for submission, especially down the line instructors may not want to require them.

**Comment by kmcnellis** [comment]
That was not actually a change, just a comma.
I'm not sure if radius is used anywhere anymore, but I need to check with the iOS app.

It is just the one dimension currently because its was the longer of the 2 (the map is horizontal)

File: server/api/submission/submission.controller.js (hunk 3 above)
Comment (kmcnellis) [inline server/api/submission/submission.controller.js:237]: Wouldn't that mess up a lot of things?

Comment (kmcnellis) [inline server/api/submission/submission.controller.js:237]: Also, this isn't currently a change to the app flow. I think that is a good feature to add later, when we can remove the requirement on location throughout the app

File: client/components/eventform/eventform.controller.js (hunk 2 above)
Comment (kmcnellis) [inline client/components/eventform/eventform.controller.js:294]: That was not actually a change, just a comma.
I'm not sure if radius is used anywhere anymore, but I need to check with the iOS app.

It is just the one dimension currently because its was the longer of the 2 (the map is horizontal)

Comment (seveibar) [inline client/components/eventform/eventform.controller.js:294]: 👍

File: server/api/submission/submission.controller.js (hunk 3 above)
Comment (seveibar) [inline server/api/submission/submission.controller.js:237]: This CANNOT be merged until this change is made.

Comment (seveibar) [inline server/api/submission/submission.controller.js:237]: (for clarification) This would break browsers/phones that upload without location, location is certainly not required it's just a validation method

Comment (kmcnellis) [inline server/api/submission/submission.controller.js:237]: I'm working on it

**Comment by kmcnellis** [comment]
Updated

File: client/components/eventform/eventform.controller.js [hunk 4]
```diff
@@ -169,31 +191,77 @@ angular.module('venueApp')
       $scope.eventInfo = event;
     };
 
+    $scope.clearMapPoly = ()=>{
+
+    };
```
Comment (seveibar) [inline client/components/eventform/eventform.controller.js]: ? any reason for this function

File: client/components/eventform/eventform.controller.js [hunk 5]
```diff
@@ -169,31 +191,77 @@ angular.module('venueApp')
...
+        var poly = [];
+        var shape = $scope.allShapes[a].overlay;
+
+        for (var b = 0; b < shape.getPath().getLength() ; b++){
+          coords.push([shape.getPath().getAt(b).lng(), shape.getPath().getAt(b).lat()]);
+        }
+        coords.push([shape.getPath().getAt(0).lng(), shape.getPath().getAt(0).lat()]);
+
+        poly.push(coords);
+
```
Comment (seveibar) [inline client/components/eventform/eventform.controller.js]: i think some comments in this section would be helpful

File: server/api/eventinfo/eventinfo.controller.js [hunk 6]
```diff
@@ -126,6 +126,48 @@ exports.create = function(req, res) {
...
+          for (var d = 0; d < coords.length; d++){
+            line.push(Number(coords[d]));
+          }
+          allline.push(line);
+        }
+        poly.push(allline);
+      }
+      shapes.push(poly);
+    }
+    evnt.location.geobounds.coordinates = shapes;
```
Comment (seveibar) [inline server/api/eventinfo/eventinfo.controller.js]: again some comments would really help

Comment (seveibar) [inline server/api/eventinfo/eventinfo.controller.js]: these a,b,c variable names are... well abc variable names

**Comment by seveibar** [comment]
again i think we need to break these PRs up- it's very hard to follow, esp. with the few comments

**Comment by seveibar** [comment]
ready to merge after the comments issue for my line notes are resolved

**Comment by kmcnellis** [comment]
Will do

**Comment by kmcnellis** [comment]
Fixed

**Comment by seveibar** [comment]
awesome thanks for the corrections
